storage/reads/aggregate_resultset.go

```diff
@@ -33,7 +33,13 @@ func NewWindowAggregateResultSet(ctx context.Context, req *datatypes.ReadWindowA
 	}
 
 	ascending := true
-	if len(req.Aggregate) > 0 && req.Aggregate[0].Type == datatypes.AggregateTypeLast {
+
+	// The following is an optimization where in the case of a single window,
+	// the selector `last` is implemented as a descending array cursor followed
+	// by a limit array cursor that selects only the first point, i.e the point
+	// with the largest timestamp, from the descending array cursor.
+	//
+	if req.Aggregate[0].Type == datatypes.AggregateTypeLast && (req.WindowEvery == 0 || req.WindowEvery == math.MaxInt64) {
```

Comment:
Are you implying it should return 0 points, not 1 point?

Reply:
Yes, because window end is exclusive.

Reply:
If my data is:
```
1, "value1"
math.MaxInt64, "value2"
```
`... |> window(every: inf) |> last()` should return `1, "value1"` not `math.MaxInt64, "value2"`

Reply:
Ah yes you are correct, this should only be checking for `req.WindowEvery == 0`.

Reply:
I think for the vast majority of the `window(every: inf)` cases, using `limit` is still more efficient? I just want to point out this one specific edge case that will not work.

Reply:
It will cause a problem only if you have `math.MaxInt64` timestamp row in your data but it is rare.

Reply:
To be honest, I don't really want to give up the reverse limit 1 optimization for `last` just because of 1 particular rare edge case, not sure what to suggest here.

Reply:
I originally used this as a reference:

https://github.com/influxdata/influxdb/blob/master/storage/reads/aggregate_resultset.go#L57-L68

But as you pointed out, this will not always be correct. That will have to be corrected as well. We also clearly need test coverage for this edge case.

Reply:
It turns out I forgot about the `range()` thing... I actually posted this https://github.com/influxdata/influxdb/pull/18414/files#r437653943 a while ago. So you can ignore my comments above.

Reply:
Ah yes, it turns out this edge case is actually invalid because influxdb's `range` requirement will always filter out any points at time `math.MaxInt64`.